pkg/controller/migmigration/hooks.go

```diff
@@ -0,0 +1,212 @@
+package migmigration
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (t *Task) runHooks() (int32, error) {
+	var hookRef *corev1.ObjectReference
+	var client k8sclient.Client
+	var err error
+
+	switch t.Phase {
+	case PreBackupHooks:
+		hookRef = t.PlanResources.MigPlan.Spec.Hooks.PreBackup
+	case PostBackupHooks:
+		hookRef = t.PlanResources.MigPlan.Spec.Hooks.PostBackup
+	case PreRestoreHooks:
+		hookRef = t.PlanResources.MigPlan.Spec.Hooks.PreRestore
+	case PostRestoreHooks:
+		hookRef = t.PlanResources.MigPlan.Spec.Hooks.PostRestore
+	default:
+		err := fmt.Errorf("Unknown Hook Phase %s", t.Phase)
+		log.Trace(err)
+		return -1, err
+	}
+
+	hook := migapi.MigHook{}
+	job := &batchv1.Job{}
+
+	if hookRef != nil {
+		err = t.Client.Get(
+			context.TODO(),
+			types.NamespacedName{
+				Name:      hookRef.Name,
+				Namespace: hookRef.Namespace,
+			},
+			&hook)
+		if err != nil {
+			log.Trace(err)
+			return -1, err
+		}
+
+		switch hook.Spec.TargetCluster {
```

Comment:
I very much think we're headed there. It might be 1.3 instead of 1.2. Part of the approach taken is

1.) as we talked on Slack there's some details about tracking status that need to be hashed out
2.) The UI will ultimately treat Hooks as reusable so that they don't need to be recreated for each plan. But there isn't time to develop the UI for 1.2 so we're talking a simpler approach to make the feature available until more time is available for UI work.